src/debug.c

```diff
@@ -1809,7 +1809,7 @@ void sigsegvHandler(int sig, siginfo_t *info, void *secret) {
         serverLog(LL_WARNING,
         "Accessing address: %p", (void*)info->si_addr);
     }
-    if (info->si_pid != -1) {
+    if (info->si_code <= 0 && info->si_pid != -1) {
```

Comment:
nitpick: should probably use `SI_FROMUSER(info->si_code)` to be more compatible and readable.

Reply:
```suggestion
    if (SI_FROMUSER(info) && info->si_pid != -1) {
```

Reply:
Looks like SI_FROMUSE isn't available (also not really documented), but I used SI_USER instead.